src/benchq/resource_estimation/default_pipelines.py

```diff
@@ -233,74 +232,55 @@ def automatic_resource_estimator(
+        algorithm_implementation (AlgorithmImplementation): The algorithm to estimate
+            resources for.
+        hardware_model (BasicArchitectureModel): The hardware model to estimate
+            resources for.
+        decoder_model (Optional[DecoderModel], optional): The decoder model to
+            run the algorithm with. Defaults to None and returns no estimate in
+            that case.
 
     Returns:
-        GraphResourceEstimator: an appropriate resource estimator for the problem
+        ResourceInfo: The resources required to run the algorithm.
     """
-    pipeline: Any = None
     assert isinstance(algorithm_implementation.program, QuantumProgram)
 
-    prev_steps = algorithm_implementation.program.steps
+    graph_size = estimate_full_graph_size(algorithm_implementation, False)
+    reduced_graph_size = estimate_full_graph_size(algorithm_implementation, True)
     algorithm_implementation.program.steps = max(DEFAULT_STEPS_TO_EXTRAPOLATE_FROM)
-    if (
-        estimate_full_graph_size(algorithm_implementation, True)
-        < LARGEST_GRAPH_TOLERANCE
-    ):
-        pipeline = partial(
-            run_fast_extrapolation_estimate,
-            steps_to_extrapolate_from=DEFAULT_STEPS_TO_EXTRAPOLATE_FROM,
-        )
-    elif (
-        estimate_full_graph_size(algorithm_implementation, False)
-        < LARGEST_GRAPH_TOLERANCE
-    ):
+
+    if graph_size < 1e7:
+        pipeline = run_precise_graph_estimate
+    elif graph_size < 1e8:
         pipeline = partial(
             run_precise_extrapolation_estimate,
             steps_to_extrapolate_from=DEFAULT_STEPS_TO_EXTRAPOLATE_FROM,
         )
-
-    algorithm_implementation.program.steps = prev_steps
-    if (
-        estimate_full_graph_size(algorithm_implementation, True)
-        < LARGEST_GRAPH_TOLERANCE
-    ):
+    elif reduced_graph_size < 1e7:
         pipeline = run_fast_graph_estimate
-    if (
-        estimate_full_graph_size(algorithm_implementation, False)
-        < LARGEST_GRAPH_TOLERANCE
-    ):
-        pipeline = run_precise_graph_estimate
-
-    if pipeline is not None:
-        return pipeline(
-            algorithm_implementation,
-            hardware_model,
-            decoder_model=decoder_model,
+    elif reduced_graph_size < 1e8:
+        pipeline = partial(
+            run_fast_extrapolation_estimate,
+            steps_to_extrapolate_from=DEFAULT_STEPS_TO_EXTRAPOLATE_FROM,
         )
     else:
-        raise ValueError(
-            "Problem size too large for resource estimation. "
-            "Try reducing the size of each of your steps in the program. "
-            "If you are creating a program from a circuit, consider breaking "
-            "the program up into steps."
-        )
+        pipeline = run_footprint_analysis_pipeline
+
+    return pipeline(
+        algorithm_implementation,
+        hardware_model,
+        decoder_model=decoder_model,
+    )
 
 
 def estimate_full_graph_size(
```

Comment:
This might be a little out of scope, but can you change `estimate_full_graph_size` to also include a count of the number of CNOT and CZ gates in the circuit as well (there should already be code for  counting the number of CNOT and CZ gates in the `QuantumProgram` object).

Reply:
Added them to the size